Block/Service/AbstractCategoriesBlockService.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Block\Service;
+
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\ClassificationBundle\Admin\CategoryAdmin;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
+use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
+use Sonata\CoreBundle\Model\Metadata;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class AbstractCategoriesBlockService.
+ *
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractCategoriesBlockService extends BaseClassificationBlockService
+{
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    /**
+     * @var CategoryAdmin
+     */
+    private $categoryAdmin;
+
+    /**
+     * @param string                   $name
+     * @param EngineInterface          $templating
+     * @param ContextManagerInterface  $contextManager
+     * @param CategoryManagerInterface $categoryManager
+     * @param CategoryAdmin            $categoryAdmin
+     */
+    public function __construct(
```

Comment:
Well in case you didn't notice, I lost [2 battles](https://github.com/sonata-project/dev-kit/issues/103) about that against @Soullivaneuh , javiereguiluz and fabpot yesterday… so just use one-line signatures, I don't want to get started on this again :disappointed: Sorry for the inconvenience.